### PR TITLE
New storage docs

### PR DESCRIPTION
**Pull request by rossriley**
This is just the first part of a work in progress but it does get us started on an overview ready for when 2.3 work starts.

### CODE REVIEW

File: source/storage/entity-metadata.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+It's important to note that this metadata driver also typehints against `Doctrine\Common\Persistence\Mapping\Driver\MappingDriver` so it is fully compatible with other Doctrine metadata drivers. More advanced applications may want to take advantage of this to bring Bolt's database metadata into a larger ecosystem.
+
+### getClassMetadata($classname)
+
+This method returns all the metadata for a given class name, for example:
+
+```
+$meta = $app['storage,etadata']->getClassMetadata('Bolt\Entity\User');
+```
```
Comment (GawainLynch) [inline source/storage/entity-metadata.md]: Typo with  `'storage,etadata'`

File: source/storage/introduction.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+## Heirarchy of Storage in Bolt
+
+There is a single point of entry for all storage interactions within Bolt and this is usually
+accessible with a call to `$app['storage']`. If you implement your own controllers, which is
+normally recommended when you want control over content fetching, then any queries that you
+would like to make can be accessed through this object.
+
+The basic heirarchy of the storage layer is described below, and clicking through gives a 
+much more indepth description of the functionality of each layer.
+
```
Comment (GawainLynch) [inline source/storage/introduction.md]: 'in depth' or 'in-depth'.

File: source/storage/repositories.md [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+
+```
+$repo = $app['storage']->getRepository('users');
+$qb = $repo->createQueryBuilder();
+```
+
+Apart from more basic queries, which can use the simpler finder methods, the primary method of querying the database is via a QueryBuilder instance. This method fetches an instance of `QueryBuilder` that is preset to select on the managed storage table.
+
+The returned instance is always an object of type `Doctrine\DBAL\Query\QueryBuilder` - much more indepth documentation for using this <a href="http://doctrine-dbal.readthedocs.org/en/latest/reference/query-builder.html">can be found here</a>.
+
```
Comment (GawainLynch) [inline source/storage/repositories.md]: 'in depth' or 'in-depth'.

**Comment by xiaohutai** [comment]
Hm, 'hierarchy' vs 'heirarchy'?

**Pull request merged**